enode.rkt

```diff
@@ -164,7 +165,9 @@
       en)))
 
 (define (dedup-children! en)
-  (set-enode-children! en (remove-duplicates (enode-children en) #:key enode-expr)))
+  (set-enode-children! en (rest (remove-duplicates
+                                 (cons en (enode-children en))
+                                 #:key enode-expr))))
 
```

Comment:
Lol

Reply:
Oh yeah, this is bug fix 5 so that the graphs actually deduplicate all the way